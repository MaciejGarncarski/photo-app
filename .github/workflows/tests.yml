name: Tests
on:
  push:
    branches:
      - main
      - develop
  pull_request: {}

jobs:
  test:
    name: 🃏 Test
    runs-on: ubuntu-latest
    steps:
      - name: 🛑 Cancel Previous Runs
        uses: styfle/cancel-workflow-action@0.11.0

      - name: ⬇️ Checkout repo
        uses: actions/checkout@v3
        with:
          persist-credentials: false      

      - name: ⎔ Set up node
        uses: actions/setup-node@v3
        with:
          node-version: lts/*

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        with:
          version: 8
          run_install: false

      - name: Get pnpm store directory
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV

      - uses: actions/cache@v3
        name: Setup pnpm cache
        with:
          path: ${{ env.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-

      - name: 📥 Download deps
        run: pnpm install
        
      - name: 🔎 Type check
        run: pnpm tsc

      - name: 🧪 Test
        run: pnpm coverage