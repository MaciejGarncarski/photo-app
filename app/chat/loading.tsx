import { Loader } from "@/components/loader/loader";

const ChatLoading = () => {
  return <Loader color="accent" size="big" marginTop />;
};

export default ChatLoading;
