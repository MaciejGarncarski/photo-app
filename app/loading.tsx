import { Loader } from '@/components/loader/loader';

const HomePageLoading = () => {
  return <Loader color="accent" size="big" marginTop />;
};

export default HomePageLoading;
