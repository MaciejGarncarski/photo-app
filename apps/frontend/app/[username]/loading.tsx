import { Loader } from "@/components/loader/loader";

const AccountLoading = () => {
  return <Loader color="accent" size="big" marginTop />;
};

export default AccountLoading;
