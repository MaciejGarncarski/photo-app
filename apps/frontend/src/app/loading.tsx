export default function Loading() {
	return <p>Loadnig...</p>
}
