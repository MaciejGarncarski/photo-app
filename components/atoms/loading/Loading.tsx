import clsx from 'clsx';

import styles from './loading.module.scss';

type LoadingVariants = 'small' | 'very-small' | 'left' | 'center' | 'no-margin';

type LoadingProps = {
  className?: string;
  variants?: Array<LoadingVariants>;
};

export const Loading = ({ variants, className }: LoadingProps) => {
  const variantClassName = variants?.map((variant) => styles[variant]);

  return (
    <div data-testid="loading" className={clsx(className, variantClassName, styles.container)}>
      <span className="visually-hidden">loading</span>
    </div>
  );
};
