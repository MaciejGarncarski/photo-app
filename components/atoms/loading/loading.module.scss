@use '/styles/common.scss';

.container {
  --loading-size: 15rem;
  height: var(--loading-size);
  width: var(--loading-size);
  margin: auto;
  text-align: center;

  border-radius: 100%;
  border: 2rem solid var(--color-primary);
  border-left-color: var(--color-secondary);
  border-right-color: var(--color-secondary);

  animation: rotation 3s infinite forwards linear;

  @include common.mediaquery('lg') {
    --loading-size: 20rem;
  }
}


@keyframes rotation {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

.small {
  --small-loading-size: 10rem;
  border-width: 1.5rem;
  height: var(--small-loading-size);
  width: var(--small-loading-size);

  @include common.mediaquery('lg') {
    --loading-size: 14rem;
  }

  @include common.mediaquery('lg') {
    --loading-size: 18rem;
  }
}

.very-small {
  --small-loading-size: 3rem;
  border-width: 0.6rem;
  height: var(--small-loading-size);
  width: var(--small-loading-size);

  @include common.mediaquery('lg') {
    --loading-size: 6rem;
  }

  @include common.mediaquery('lg') {
    --loading-size: 7rem;
  }
}

.left {
  margin: auto;
  margin-left: 0;
}

.center {
  margin: 0;
}