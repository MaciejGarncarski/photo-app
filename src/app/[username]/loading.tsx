import { Loader } from '@/src/components/loader/loader';

const Loading = () => {
  return <Loader marginTop color="blue" size="normal" />;
};

export default Loading;
