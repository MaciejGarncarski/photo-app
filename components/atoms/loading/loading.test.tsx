import { Loading } from '@/components/atoms/loading/Loading';

import { render } from '@/tests';

describe('<Loading />', () => {
  it('should render default variant', () => {
    const { getByTestId } = render(<Loading />);

    const loading = getByTestId('loading');
    expect(loading).toBeInTheDocument();
    expect(loading).toMatchSnapshot();
  });

  it('should render small variant', () => {
    const { getByTestId } = render(<Loading variants={['small']} />);

    const loading = getByTestId('loading');
    expect(loading).toBeInTheDocument();
    expect(loading).toMatchSnapshot();
    expect(loading).toHaveClass('small');
  });

  it('should render center variant', () => {
    const { getByTestId } = render(<Loading variants={['center']} />);

    const loading = getByTestId('loading');
    expect(loading).toBeInTheDocument();
    expect(loading).toMatchSnapshot();
    expect(loading).toHaveClass('center');
  });

  it('should render multiple variants', () => {
    const { getByTestId } = render(<Loading variants={['center', 'very-small']} />);

    const loading = getByTestId('loading');
    expect(loading).toBeInTheDocument();
    expect(loading).toMatchSnapshot();
    expect(loading).toHaveClass('center', 'very-small');
  });
});
