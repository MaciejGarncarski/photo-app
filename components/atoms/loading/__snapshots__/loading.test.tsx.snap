// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render center variant 1`] = `
<div
  class="center container"
  data-testid="loading"
>
  <span
    class="visually-hidden"
  >
    loading
  </span>
</div>
`;

exports[`<Loading /> should render default variant 1`] = `
<div
  class="container"
  data-testid="loading"
>
  <span
    class="visually-hidden"
  >
    loading
  </span>
</div>
`;

exports[`<Loading /> should render multiple variants 1`] = `
<div
  class="center very-small container"
  data-testid="loading"
>
  <span
    class="visually-hidden"
  >
    loading
  </span>
</div>
`;

exports[`<Loading /> should render small variant 1`] = `
<div
  class="small container"
  data-testid="loading"
>
  <span
    class="visually-hidden"
  >
    loading
  </span>
</div>
`;
